src/black/mode.py

```diff
@@ -147,5 +148,6 @@ def get_cache_key(self) -> str:
             str(int(self.is_ipynb)),
             str(int(self.magic_trailing_comma)),
             str(int(self.experimental_string_processing)),
+            ",".join(sorted(self.python_cell_magics)),
```

Comment:
At this point I'm a little worried that we might hit some filesystem file name length limits. I assume that Python cell magics are also filesystem safe? can't contain slashes, colons, whatever isn't safe.

The cache should fail gracefully but I'm not entirely sure if it does so yeah. This can be resolved separately though.

Reply:
Good point. Maybe we can hash the value with some consistent hashing algorithm available in the stdlib.

Reply:
It doesn't need to be human-readable, though it's helpful for debugging if it's not too bad. The other components of the key should mostly just be ints, so I think we can get away with not hashing them.

Reply:
Could you make the change to hashing? Also, there are a few merge conflicts now.

Reply:
Of course.  I've done that and resolved the merge conflicts.